File: Java/Concatenate.java
public void concatenate(Stack otherStack) {
    if (top + otherStack.top < size - 1) {
        for (int i = 0; i <= otherStack.top; i++) {
            stack[top + i + 1] = otherStack.stack[i];
        }
        top += otherStack.top + 1;
        otherStack.top = -1; // Clear the other stack
        otherStack.clear();
    } else {
        System.out.println("Destination stack is not large enough to concatenate.");
    }
}

File: Python/Concatenate.py
def concatenate(self, other_stack):
    if len(self.stack) + len(other_stack.stack) <= self.size:
        self.stack.extend(other_stack.stack)
        self.top = len(self.stack) - 1
        other_stack.clear()
    else:
        print("Destination stack is not large enough to concatenate.")
